### PR TITLE
NOISSUE - Fix Provision CLI (Clients)

### PR DESCRIPTION
**Pull request by rodneyosodo**
### What does this do?
Fix provision command in CLI with respect to clients

### Which issue(s) does this PR fix/relate to?
N/A

### List any changes that modify/break current functionality
It creates things and channels with enabled status

### Have you included tests for your changes?
No

### Did you document any new/modified functionality?
No

### Notes
N/A

### CODE REVIEW

File: api/openapi/users.yml [hunk 1]
```diff
@@ -1,117 +1,290 @@
...
+    externalDocs:
+      description: Find out more
+      url: http://swagger.io
+  - name: Policies
+    description: Access to user policies
+    externalDocs:
+      description: Find out more about our store
+      url: http://swagger.io
 
 paths:
```
Comment (SammyOina) [inline api/openapi/users.yml]: good practice to set operation IDs

File: api/openapi/users.yml
Comment (SammyOina) [inline api/openapi/users.yml:1]: issue token request is wrong

File: cmd/smpp-notifier/main.go [hunk 2]
```diff
@@ -125,8 +124,8 @@ func main() {
 
 }
 
-func newService(db *sqlx.DB, tracer opentracing.Tracer, auth policies.AuthServiceClient, c config, sc mfsmpp.Config, logger mflog.Logger) notifiers.Service {
-	database := notifierPg.NewDatabase(db)
+func newService(db *sqlx.DB, tracer trace.Tracer, auth policies.AuthServiceClient, c config, sc mfsmpp.Config, logger mflog.Logger) notifiers.Service {
```
Comment (SammyOina) [inline cmd/smpp-notifier/main.go]: pass s, sc and ec by pointer

File: cmd/smtp-notifier/main.go [hunk 3]
```diff
@@ -125,8 +124,8 @@ func main() {
 
 }
 
-func newService(db *sqlx.DB, tracer opentracing.Tracer, auth policies.AuthServiceClient, c config, ec email.Config, logger mflog.Logger) notifiers.Service {
-	database := notifierPg.NewDatabase(db)
+func newService(db *sqlx.DB, tracer trace.Tracer, auth policies.AuthServiceClient, c config, ec email.Config, logger mflog.Logger) notifiers.Service {
```
Comment (SammyOina) [inline cmd/smtp-notifier/main.go]: pass c and ec by pointer

File: cmd/users/main.go [hunk 4]
```diff
@@ -156,26 +150,6 @@ func main() {
...
-			semconv.ServiceNameKey.String(svcName),
-		)),
-	)
-	otel.SetTracerProvider(tp)
-	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}))
-
-	return tp, nil
-}
-
 func newService(db *sqlx.DB, tracer trace.Tracer, c config, ec email.Config, logger mflog.Logger) (clients.Service, groups.GroupService, policies.PolicyService) {
```
Comment (SammyOina) [inline cmd/users/main.go]: pass c and ec by pointer

File: consumers/notifiers/api/transport.go [hunk 5]
```diff
@@ -32,36 +31,36 @@ const (
 )
 
 // MakeHandler returns a HTTP handler for API endpoints.
-func MakeHandler(svc notifiers.Service, tracer opentracing.Tracer, logger logger.Logger) http.Handler {
+func MakeHandler(svc notifiers.Service, logger logger.Logger) http.Handler {
```
Comment (SammyOina) [inline consumers/notifiers/api/transport.go]: import shadow for logger consider renaming

File: consumers/notifiers/api/endpoint_test.go [hunk 6]
```diff
@@ -77,7 +76,7 @@ func newService(tokens map[string]string) notifiers.Service {
 
 func newServer(svc notifiers.Service) *httptest.Server {
 	logger := logger.NewMock()
```
Comment (SammyOina) [inline consumers/notifiers/api/endpoint_test.go]: import shadow

File: http/api/transport.go [hunk 7]
```diff
@@ -42,21 +41,21 @@ var (
 var channelPartRegExp = regexp.MustCompile(`^/channels/([\w\-]+)/messages(/[^?]*)?(\?.*)?$`)
 
 // MakeHandler returns a HTTP handler for API endpoints.
-func MakeHandler(svc adapter.Service, tracer opentracing.Tracer, logger logger.Logger) http.Handler {
+func MakeHandler(svc adapter.Service, logger logger.Logger) http.Handler {
```
Comment (SammyOina) [inline http/api/transport.go]: import shadow

File: http/api/endpoint_test.go [hunk 8]
```diff
@@ -29,7 +27,7 @@ func newService(cc policies.ThingsServiceClient) adapter.Service {
 
 func newHTTPServer(svc adapter.Service) *httptest.Server {
 	logger := logger.NewMock()
```
Comment (SammyOina) [inline http/api/endpoint_test.go]: import shadow

File: pkg/sdk/go/policies.go [hunk 9]
```diff
@@ -26,46 +26,36 @@ type Policy struct {
 // AddPolicy creates a policy for the given subject, so that, after
 // AddPolicy, `subject` has a `relation` on `object`. Returns a non-nil
 // error in case of failures.
-func (sdk mfSDK) AddPolicy(p Policy, token string) (Policy, errors.SDKError) {
+func (sdk mfSDK) AddPolicy(p Policy, token string) errors.SDKError {
```
Comment (SammyOina) [inline pkg/sdk/go/policies.go]: pass sdk and policy as pointers

File: pkg/sdk/go/policies.go [hunk 10]
```diff
@@ -26,46 +26,36 @@ type Policy struct {
...
+		return sdkerr
 	}
 
-	return p, nil
+	return nil
 }
 
 // UpdatePolicy updates policies based on the given policy structure.
-func (sdk mfSDK) UpdatePolicy(p Policy, token string) (Policy, errors.SDKError) {
+func (sdk mfSDK) UpdatePolicy(p Policy, token string) errors.SDKError {
```
Comment (SammyOina) [inline pkg/sdk/go/policies.go]: pass sdk and policy as pointers

File: pkg/sdk/go/groups_test.go [hunk 11]
```diff
@@ -0,0 +1,779 @@
...
+	"github.com/mainflux/mainflux/internal/testsutil"
+	"github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/pkg/errors"
+	sdk "github.com/mainflux/mainflux/pkg/sdk/go"
+	"github.com/mainflux/mainflux/users/clients"
+	cmocks "github.com/mainflux/mainflux/users/clients/mocks"
+	"github.com/mainflux/mainflux/users/groups"
+	"github.com/mainflux/mainflux/users/groups/api"
+	"github.com/mainflux/mainflux/users/groups/mocks"
+	gmocks "github.com/mainflux/mainflux/users/groups/mocks"
```
Comment (SammyOina) [inline pkg/sdk/go/groups_test.go]: repeated import

File: pkg/sdk/go/groups_test.go [hunk 12]
```diff
@@ -0,0 +1,779 @@
...
+	"github.com/mainflux/mainflux/users/jwt"
+	pmocks "github.com/mainflux/mainflux/users/policies/mocks"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+func newGroupsServer(svc groups.Service) *httptest.Server {
+	logger := logger.NewMock()
```
Comment (SammyOina) [inline pkg/sdk/go/groups_test.go]: import shadow

File: pkg/sdk/go/message_test.go [hunk 13]
```diff
@@ -30,7 +29,7 @@ func newMessageService(cc policies.ThingsServiceClient) adapter.Service {
 
 func newMessageServer(svc adapter.Service) *httptest.Server {
 	logger := logger.NewMock()
```
Comment (SammyOina) [inline pkg/sdk/go/message_test.go]: import shadow

File: pkg/sdk/go/setup_test.go [hunk 14]
```diff
@@ -153,7 +143,7 @@ func convertMembershipsPage(m sdk.MembershipsPage) groups.MembershipsPage {
 
 func convertClientPage(p sdk.PageMetadata) clients.Page {
```
Comment (SammyOina) [inline pkg/sdk/go/setup_test.go]: pass p as pointer

File: pkg/sdk/go/setup_test.go [hunk 15]
```diff
@@ -211,18 +170,9 @@ func convertMemberships(gs []sdk.Group) []groups.Group {
...
-func convertMembers(gs []sdk.User) []clients.Client {
-	cg := []clients.Client{}
-	for _, g := range gs {
-		cg = append(cg, convertClient(g))
-	}
-
-	return cg
-}
-
 func convertGroup(g sdk.Group) groups.Group {
```
Comment (SammyOina) [inline pkg/sdk/go/setup_test.go]: pass g as pointer

File: pkg/sdk/go/setup_test.go [hunk 16]
```diff
@@ -261,7 +211,7 @@ func convertChildren(gs []*sdk.Group) []*groups.Group {
 
 func convertClient(c sdk.User) clients.Client {
```
Comment (SammyOina) [inline pkg/sdk/go/setup_test.go]: pass c as pointer

File: things/policies/service.go [hunk 17]
```diff
@@ -55,8 +55,11 @@ func (svc service) Authorize(ctx context.Context, entityType string, p Policy) e
 }
 
 func (svc service) AuthorizeByKey(ctx context.Context, entityType string, p Policy) (string, error) {
```
Comment (SammyOina) [inline things/policies/service.go]: pass svc and p as pointers

File: things/policies/service.go [hunk 18]
```diff
@@ -153,3 +156,15 @@ func (svc service) checkActionRank(ctx context.Context, clientID string, p Polic
 	return apiutil.ErrHigherPolicyRank
 
 }
+
+func (svc service) hasThing(ctx context.Context, p Policy) (string, error) {
```
Comment (SammyOina) [inline things/policies/service.go]: pass p ad svc as pointers

File: twins/api/http/transport.go [hunk 19]
```diff
@@ -32,50 +31,50 @@ const (
 )
 
 // MakeHandler returns a HTTP handler for API endpoints.
-func MakeHandler(tracer opentracing.Tracer, svc twins.Service, logger logger.Logger) http.Handler {
+func MakeHandler(svc twins.Service, logger logger.Logger) http.Handler {
 	opts := []kithttp.ServerOption{
 		kithttp.ServerErrorEncoder(apiutil.LoggingErrorEncoder(logger, encodeError)),
```
Comment (SammyOina) [inline twins/api/http/transport.go]: import shadow on logger

File: twins/api/http/endpoint_twins_test.go [hunk 20]
```diff
@@ -85,7 +84,7 @@ func (tr testRequest) make() (*http.Response, error) {
 
 func newServer(svc twins.Service) *httptest.Server {
 	logger := logger.NewMock()
```
Comment (SammyOina) [inline twins/api/http/endpoint_twins_test.go]: import shadow

File: twins/tracing/states.go [hunk 21]
```diff
@@ -20,55 +20,50 @@ const (
...
-	span := createSpan(ctx, trm.tracer, updateStateOp)
-	defer span.Finish()
-	ctx = opentracing.ContextWithSpan(ctx, span)
+	ctx, span := createSpan(ctx, trm.tracer, updateStateOp)
+	defer span.End()
 
 	return trm.repo.Update(ctx, st)
 }
 
 func (trm stateRepositoryMiddleware) Count(ctx context.Context, tw twins.Twin) (int64, error) {
```
Comment (SammyOina) [inline twins/tracing/states.go]: pass tw as pointer

File: twins/tracing/twins.go [hunk 22]
```diff
@@ -23,125 +23,109 @@ const (
...
 // TwinRepositoryMiddleware tracks request and their latency, and adds spans to context.
-func TwinRepositoryMiddleware(tracer opentracing.Tracer, repo twins.TwinRepository) twins.TwinRepository {
+func TwinRepositoryMiddleware(tracer trace.Tracer, repo twins.TwinRepository) twins.TwinRepository {
 	return twinRepositoryMiddleware{
 		tracer: tracer,
 		repo:   repo,
 	}
 }
 
 func (trm twinRepositoryMiddleware) Save(ctx context.Context, tw twins.Twin) (string, error) {
```
Comment (SammyOina) [inline twins/tracing/twins.go]: pass tw as pointer

File: twins/tracing/twins.go [hunk 23]
```diff
@@ -23,125 +23,109 @@ const (
...
-	span := createSpan(ctx, trm.tracer, saveTwinOp)
-	defer span.Finish()
-	ctx = opentracing.ContextWithSpan(ctx, span)
+	ctx, span := createSpan(ctx, trm.tracer, saveTwinOp)
+	defer span.End()
 
 	return trm.repo.Save(ctx, tw)
 }
 
 func (trm twinRepositoryMiddleware) Update(ctx context.Context, tw twins.Twin) error {
```
Comment (SammyOina) [inline twins/tracing/twins.go]: pass tw as pointer

File: twins/tracing/twins.go [hunk 24]
```diff
@@ -23,125 +23,109 @@ const (
...
 // TwinCacheMiddleware tracks request and their latency, and adds spans to context.
-func TwinCacheMiddleware(tracer opentracing.Tracer, cache twins.TwinCache) twins.TwinCache {
+func TwinCacheMiddleware(tracer trace.Tracer, cache twins.TwinCache) twins.TwinCache {
 	return twinCacheMiddleware{
 		tracer: tracer,
 		cache:  cache,
 	}
 }
 
 func (tcm twinCacheMiddleware) Save(ctx context.Context, twin twins.Twin) error {
```
Comment (SammyOina) [inline twins/tracing/twins.go]: pass tw as pointer

File: twins/tracing/twins.go [hunk 25]
```diff
@@ -23,125 +23,109 @@ const (
...
-	span := createSpan(ctx, tcm.tracer, saveTwinsOp)
-	defer span.Finish()
-	ctx = opentracing.ContextWithSpan(ctx, span)
+	ctx, span := createSpan(ctx, tcm.tracer, saveTwinsOp)
+	defer span.End()
 
 	return tcm.cache.SaveIDs(ctx, channel, subtopic, ids)
 }
 
 func (tcm twinCacheMiddleware) Update(ctx context.Context, twin twins.Twin) error {
```
Comment (SammyOina) [inline twins/tracing/twins.go]: pass twin as pointer

File: users/clients/api/logging.go [hunk 26]
```diff
@@ -23,7 +23,7 @@ func LoggingMiddleware(svc clients.Service, logger log.Logger) clients.Service {
 
 func (lm *loggingMiddleware) RegisterClient(ctx context.Context, token string, client clients.Client) (c clients.Client, err error) {
```
Comment (SammyOina) [inline users/clients/api/logging.go]: pass client as pointer

File: users/clients/api/logging.go [hunk 27]
```diff
@@ -83,7 +83,7 @@ func (lm *loggingMiddleware) ViewProfile(ctx context.Context, token string) (c c
 
 func (lm *loggingMiddleware) ListClients(ctx context.Context, token string, pm clients.Page) (cp clients.ClientsPage, err error) {
```
Comment (SammyOina) [inline users/clients/api/logging.go]: pass pm as pointer

File: users/clients/api/logging.go [hunk 28]
```diff
@@ -95,7 +95,7 @@ func (lm *loggingMiddleware) ListClients(ctx context.Context, token string, pm c
 
 func (lm *loggingMiddleware) UpdateClient(ctx context.Context, token string, client clients.Client) (c clients.Client, err error) {
```
Comment (SammyOina) [inline users/clients/api/logging.go]: pass client as pointer

File: users/clients/api/logging.go [hunk 29]
```diff
@@ -107,7 +107,7 @@ func (lm *loggingMiddleware) UpdateClient(ctx context.Context, token string, cli
 
 func (lm *loggingMiddleware) UpdateClientTags(ctx context.Context, token string, client clients.Client) (c clients.Client, err error) {
```
Comment (SammyOina) [inline users/clients/api/logging.go]: pass client as pointer

File: users/clients/api/logging.go [hunk 30]
```diff
@@ -178,7 +178,7 @@ func (lm *loggingMiddleware) SendPasswordReset(ctx context.Context, host, email,
 
 func (lm *loggingMiddleware) UpdateClientOwner(ctx context.Context, token string, client clients.Client) (c clients.Client, err error) {
```
Comment (SammyOina) [inline users/clients/api/logging.go]: pass client as pointer

File: users/clients/api/logging.go [hunk 31]
```diff
@@ -214,7 +214,7 @@ func (lm *loggingMiddleware) DisableClient(ctx context.Context, token string, id
 
 func (lm *loggingMiddleware) ListMembers(ctx context.Context, token, groupID string, cp clients.Page) (mp clients.MembersPage, err error) {
```
Comment (SammyOina) [inline users/clients/api/logging.go]: pass cp as pointer

File: users/clients/api/logging.go [hunk 32]
```diff
@@ -190,7 +190,7 @@ func (lm *loggingMiddleware) UpdateClientOwner(ctx context.Context, token string
 
 func (lm *loggingMiddleware) EnableClient(ctx context.Context, token string, id string) (c clients.Client, err error) {
```
Comment (SammyOina) [inline users/clients/api/logging.go]:
```suggestion
func (lm *loggingMiddleware) EnableClient(ctx context.Context, token, id string) (c clients.Client, err error) {
```

File: users/clients/api/logging.go [hunk 33]
```diff
@@ -202,7 +202,7 @@ func (lm *loggingMiddleware) EnableClient(ctx context.Context, token string, id
 
 func (lm *loggingMiddleware) DisableClient(ctx context.Context, token string, id string) (c clients.Client, err error) {
```
Comment (SammyOina) [inline users/clients/api/logging.go]:
```suggestion
func (lm *loggingMiddleware) DisableClient(ctx context.Context, token, id string) (c clients.Client, err error) {
```

File: users/clients/postgres/clients.go [hunk 34]
```diff
@@ -224,26 +237,7 @@ func (repo clientRepo) Update(ctx context.Context, client clients.Client) (clien
...
-	var rClient dbClient
-	if err := row.StructScan(&rClient); err != nil {
-		return clients.Client{}, err
-	}
-
-	return toClient(rClient)
+	return repo.update(ctx, client, q)
 }
 
 func (repo clientRepo) UpdateTags(ctx context.Context, client clients.Client) (clients.Client, error) {
```
Comment (SammyOina) [inline users/clients/postgres/clients.go]: pass client as pointer

File: users/clients/postgres/clients.go [hunk 35]
```diff
@@ -252,25 +246,7 @@ func (repo clientRepo) UpdateTags(ctx context.Context, client clients.Client) (c
...
-	var rClient dbClient
-	if err := row.StructScan(&rClient); err != nil {
-		return clients.Client{}, err
-	}
-
-	return toClient(rClient)
+	return repo.update(ctx, client, q)
 }
 
 func (repo clientRepo) UpdateIdentity(ctx context.Context, client clients.Client) (clients.Client, error) {
```
Comment (SammyOina) [inline users/clients/postgres/clients.go]: pass client as pointer

File: users/clients/postgres/clients.go [hunk 36]
```diff
@@ -279,25 +255,7 @@ func (repo clientRepo) UpdateIdentity(ctx context.Context, client clients.Client
...
-	var rClient dbClient
-	if err := row.StructScan(&rClient); err != nil {
-		return clients.Client{}, err
-	}
-
-	return toClient(rClient)
+	return repo.update(ctx, client, q)
 }
 
 func (repo clientRepo) UpdateSecret(ctx context.Context, client clients.Client) (clients.Client, error) {
```
Comment (SammyOina) [inline users/clients/postgres/clients.go]: pass client as pointer

File: users/clients/postgres/clients.go [hunk 37]
```diff
@@ -306,25 +264,7 @@ func (repo clientRepo) UpdateSecret(ctx context.Context, client clients.Client)
...
-	var rClient dbClient
-	if err := row.StructScan(&rClient); err != nil {
-		return clients.Client{}, err
-	}
-
-	return toClient(rClient)
+	return repo.update(ctx, client, q)
 }
 
 func (repo clientRepo) UpdateOwner(ctx context.Context, client clients.Client) (clients.Client, error) {
```
Comment (SammyOina) [inline users/clients/postgres/clients.go]: pass client as pointer

File: users/clients/postgres/clients.go [hunk 38]
```diff
@@ -333,35 +273,24 @@ func (repo clientRepo) UpdateOwner(ctx context.Context, client clients.Client) (
...
         RETURNING id, name, tags, identity, metadata, COALESCE(owner, '') AS owner, status, created_at, updated_at`, status)
 
-	dbc := dbClient{
-		ID: id,
+	client := clients.Client{ID: id}
+
+	return repo.update(ctx, client, q)
+}
+
+func (repo clientRepo) update(ctx context.Context, client clients.Client, query string) (clients.Client, error) {
```
Comment (SammyOina) [inline users/clients/postgres/clients.go]: pass client as pointer

File: users/clients/service.go [hunk 39]
```diff
@@ -147,43 +146,43 @@ func (svc service) ViewClient(ctx context.Context, token string, id string) (Cli
...
 
-	if err := svc.policies.Evaluate(ctx, entityType, policies.Policy{Subject: ir.ID, Object: id, Actions: []string{listRelationKey}}); err != nil {
+	if err := svc.policies.Evaluate(ctx, entityType, policies.Policy{Subject: ir, Object: id, Actions: []string{listRelationKey}}); err != nil {
 		return Client{}, err
 	}
 
 	return svc.clients.RetrieveByID(ctx, id)
 }
 
 func (svc service) ViewProfile(ctx context.Context, token string) (Client, error) {
```
Comment (SammyOina) [inline users/clients/service.go]: pass svc as pointer

File: users/clients/service.go [hunk 40]
```diff
@@ -147,43 +146,43 @@ func (svc service) ViewClient(ctx context.Context, token string, id string) (Cli
...
-	ir, err := svc.Identify(ctx, token)
+	id, err := svc.Identify(ctx, token)
 	if err != nil {
 		return Client{}, err
 	}
-	return svc.clients.RetrieveByID(ctx, ir.ID)
+	return svc.clients.RetrieveByID(ctx, id)
 }
 
 func (svc service) ListClients(ctx context.Context, token string, pm Page) (ClientsPage, error) {
```
Comment (SammyOina) [inline users/clients/service.go]: pass svc and pm by pointer

File: users/clients/service.go [hunk 41]
```diff
@@ -250,11 +249,11 @@ func (svc service) GenerateResetToken(ctx context.Context, email, host string) e
 }
 
 func (svc service) ResetSecret(ctx context.Context, resetToken, secret string) error {
```
Comment (SammyOina) [inline users/clients/service.go]: pass svc as pointer

File: users/clients/service.go [hunk 42]
```diff
@@ -281,14 +280,14 @@ func (svc service) ResetSecret(ctx context.Context, resetToken, secret string) e
 }
 
 func (svc service) UpdateClientSecret(ctx context.Context, token, oldSecret, newSecret string) (Client, error) {
```
Comment (SammyOina) [inline users/clients/service.go]: pass svc as pointer

File: users/clients/service.go [hunk 43]
```diff
@@ -347,15 +346,15 @@ func (svc service) DisableClient(ctx context.Context, token, id string) (Client,
 }
 
 func (svc service) ListMembers(ctx context.Context, token, groupID string, pm Page) (MembersPage, error) {
```
Comment (SammyOina) [inline users/clients/service.go]: pass svc  by pointer

Comment (SammyOina) [reply users/clients/service.go]: pm as well

File: users/clients/service.go [hunk 44]
```diff
@@ -377,25 +376,25 @@ func (svc service) authorize(ctx context.Context, entityType string, p policies.
...
+	if err = svc.policies.CheckAdmin(ctx, id); err == nil {
 		return nil
 	}
-	p.Subject = ir.ID
+	p.Subject = id
 	return svc.policies.Evaluate(ctx, entityType, p)
 }
 
-func (svc service) Identify(ctx context.Context, tkn string) (UserIdentity, error) {
+func (svc service) Identify(ctx context.Context, tkn string) (string, error) {
```
Comment (SammyOina) [inline users/clients/service.go]: pass svc by pointer

File: users/groups/api/logging.go [hunk 45]
```diff
@@ -20,9 +20,9 @@ func LoggingMiddleware(svc groups.Service, logger log.Logger) groups.Service {
 	return &loggingMiddleware{logger, svc}
 }
 
-func (lm *loggingMiddleware) CreateGroup(ctx context.Context, token string, group groups.Group) (rGroup groups.Group, err error) {
+func (lm *loggingMiddleware) CreateGroup(ctx context.Context, token string, group groups.Group) (g groups.Group, err error) {
```
Comment (SammyOina) [inline users/groups/api/logging.go]: pass groupp as pointer

File: users/groups/api/logging.go [hunk 46]
```diff
@@ -32,9 +32,9 @@ func (lm *loggingMiddleware) CreateGroup(ctx context.Context, token string, grou
 	return lm.svc.CreateGroup(ctx, token, group)
 }
 
-func (lm *loggingMiddleware) UpdateGroup(ctx context.Context, token string, group groups.Group) (rGroup groups.Group, err error) {
+func (lm *loggingMiddleware) UpdateGroup(ctx context.Context, token string, group groups.Group) (g groups.Group, err error) {
```
Comment (SammyOina) [inline users/groups/api/logging.go]: pass group as pointer

File: users/groups/api/logging.go [hunk 47]
```diff
@@ -58,7 +58,7 @@ func (lm *loggingMiddleware) ViewGroup(ctx context.Context, token, id string) (g
 
 func (lm *loggingMiddleware) ListGroups(ctx context.Context, token string, gp groups.GroupsPage) (cg groups.GroupsPage, err error) {
```
Comment (SammyOina) [inline users/groups/api/logging.go]: pass gp as pointer

File: users/groups/api/logging.go [hunk 48]
```diff
@@ -70,7 +70,7 @@ func (lm *loggingMiddleware) ListGroups(ctx context.Context, token string, gp gr
 
 func (lm *loggingMiddleware) EnableGroup(ctx context.Context, token string, id string) (g groups.Group, err error) {
```
Comment (SammyOina) [inline users/groups/api/logging.go]:
```suggestion
func (lm *loggingMiddleware) EnableGroup(ctx context.Context, token, id string) (g groups.Group, err error) {
```

File: users/groups/api/logging.go [hunk 49]
```diff
@@ -82,7 +82,7 @@ func (lm *loggingMiddleware) EnableGroup(ctx context.Context, token string, id s
 
 func (lm *loggingMiddleware) DisableGroup(ctx context.Context, token string, id string) (g groups.Group, err error) {
```
Comment (SammyOina) [inline users/groups/api/logging.go]:
```suggestion
func (lm *loggingMiddleware) DisableGroup(ctx context.Context, token, id string) (g groups.Group, err error) {
```

File: users/groups/api/requests.go [hunk 50]
```diff
@@ -12,6 +12,9 @@ type createGroupReq struct {
 }
 
 func (req createGroupReq) validate() error {
```
Comment (SammyOina) [inline users/groups/api/requests.go]: pass req as pointer

File: users/groups/api/requests.go [hunk 51]
```diff
@@ -48,6 +52,9 @@ type listGroupsReq struct {
 }
 
 func (req listGroupsReq) validate() error {
```
Comment (SammyOina) [inline users/groups/api/requests.go]: pass req as pointer

File: users/groups/tracing/tracing.go [hunk 52]
```diff
@@ -19,12 +19,6 @@ func TracingMiddleware(gsvc groups.Service, tracer trace.Tracer) groups.Service
 	return &tracingMiddleware{tracer, gsvc}
 }
 
-func (tm *tracingMiddleware) ListMemberships(ctx context.Context, token, clientID string, gm groups.GroupsPage) (groups.MembershipsPage, error) {
-	ctx, span := tm.tracer.Start(ctx, "svc_list_memberships")
-	defer span.End()
-	return tm.gsvc.ListMemberships(ctx, token, clientID, gm)
-}
-
 func (tm *tracingMiddleware) CreateGroup(ctx context.Context, token string, g groups.Group) (groups.Group, error) {
```
Comment (SammyOina) [inline users/groups/tracing/tracing.go]: pass g as pointer

File: users/groups/tracing/tracing.go [hunk 53]
```diff
@@ -49,6 +43,12 @@ func (tm *tracingMiddleware) ListGroups(ctx context.Context, token string, gm gr
 
 }
 
+func (tm *tracingMiddleware) ListMemberships(ctx context.Context, token, clientID string, gm groups.GroupsPage) (groups.MembershipsPage, error) {
```
Comment (SammyOina) [inline users/groups/tracing/tracing.go]: pass gm as pointer

File: users/groups/tracing/tracing.go [hunk 54]
```diff
@@ -49,6 +43,12 @@ func (tm *tracingMiddleware) ListGroups(ctx context.Context, token string, gm gr
...
+	ctx, span := tm.tracer.Start(ctx, "svc_list_memberships")
+	defer span.End()
+	return tm.gsvc.ListMemberships(ctx, token, clientID, gm)
+}
+
 func (tm *tracingMiddleware) UpdateGroup(ctx context.Context, token string, g groups.Group) (groups.Group, error) {
```
Comment (SammyOina) [inline users/groups/tracing/tracing.go]: pass g as pointer

File: users/policies/service.go [hunk 55]
```diff
@@ -105,9 +105,9 @@ func (svc service) DeletePolicy(ctx context.Context, token string, p Policy) err
 }
 
 func (svc service) ListPolicy(ctx context.Context, token string, pm Page) (PolicyPage, error) {
```
Comment (SammyOina) [inline users/policies/service.go]: pass pm as pointer

**Review by SammyOina — CHANGES_REQUESTED** [review]

File: api/openapi/users.yml
Comment (rodneyosodo) [reply api/openapi/users.yml:1]: Will be solved by https://github.com/mainflux/mainflux/pull/1741

**Review by dborovcanin — CHANGES_REQUESTED** [review]
@rodneyosodo Please resolve conflicts so we can review this one.

File: api/openapi/users.yml (hunk 1 above)
Comment (ghost) [reply api/openapi/users.yml]: They are optional - https://swagger.io/docs/specification/paths-and-operations/

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**